Add support for Secret or ConfigMap based environment variables in `ContainerTemplate` (#10623)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/con-configuration-points-configmaps.adoc

To use a ConfigMap for metrics configuration, you add a reference to the `metricsConfig` configuration of the component in the same way.

> `template` properties make data from a ConfigMap (or Secret) mounted to a pod available as environment variables or volumes.

Reviewer: ```suggestion
`template` properties allow data from a `ConfigMap` or `Secret` to be mounted in a pod as environment variables or volumes.
```
